servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -331,25 +330,15 @@ public String getContextPath() {
         return delegate.getContextPath();
     }
 
+    @SuppressWarnings("resource")
     @Override
     public InputStream getInputStream() throws IOException {
-        CompletableFuture<? extends CloseableAvailableByteBody> buffered = byteBody.buffered.get();
-        if (buffered == null) {
-            return delegate.getInputStream();
-        } else {
-            try (CloseableAvailableByteBody split = buffered.get().split()) {
-                return split.toInputStream();
-            } catch (InterruptedException e) {
-                throw new InterruptedIOException();
-            } catch (ExecutionException e) {
-                throw new IOException(e);
-            }
-        }
+        return byteBody().split(ByteBody.SplitBackpressureMode.FASTEST).toInputStream();
```

Comment:
do we need to split the stream for this use case? Can't it just call `delegate.getInputStream()`?

Reply:
If e.g. a filter has already read the body, delegate.getInputStream will be empty